pkg/resolver/resolver.go

```diff
@@ -71,7 +71,18 @@ func (r *Resolver) Resolve(reqs *chartutil.Requirements, repoNames map[string]st
 			return nil, fmt.Errorf("dependency %q has an invalid version/constraint format: %s", d.Name, err)
 		}
 
-		repoIndex, err := repo.LoadIndexFile(r.helmhome.CacheIndex(repoNames[d.Name]))
+		// repo does not exist in cache but has url info
+		cacheRepoName := repoNames[d.Name]
```

Comment:
Nit: The if statement could be simplified to
```suggestion
		cacheRepoName, ok := repoNames[d.Name]
		if !ok && d.Repository != "" {
```

Reply:
Since this is just a minor, non-blocking nit. I am going to merge